ext/mysqli/mysqli_api.c

```diff
@@ -1679,15 +1679,18 @@ PHP_FUNCTION(mysqli_options)
 {
 	MY_MYSQL		*mysql;
 	zval			*mysql_link = NULL;
-	zval			*mysql_value;
+	zend_string		*mysql_value_str;
+	zend_long		mysql_value_long;
 	zend_long			mysql_option;
-	unsigned int	l_value;
 	zend_long			ret;
 	int				expected_type;
 
-	if (zend_parse_method_parameters(ZEND_NUM_ARGS(), getThis(), "Olz", &mysql_link, mysqli_link_class_entry, &mysql_option, &mysql_value) == FAILURE) {
-		RETURN_THROWS();
-	}
+	ZEND_PARSE_PARAMETERS_START(3, 3)
```

Comment:
This is not right, because this uses parse_method_parameters.

Reply:
So the best solution with fast ZPP is to do a conditional ZPP based on `if (getThis())`?